plugin/core/transports.py

```diff
@@ -136,8 +171,8 @@ def __del__(self) -> None:
 
     def _read_loop(self) -> None:
         try:
-            while self._reader:
-                payload = self._processor.read_data(self._reader)
+            while True:
```

Comment:
I tried to Google this and also replicate it in REPL with `io.BytesIO`, but it seems that streams are always truthy?Or am I taking down a Chesterton's fence here?

Reply:
@predragnikolic Do you know anything about this? Do `self._reader`/`self._writer` ever become falsey?

Reply:
> Do you know anything about this? 

No, I don't know, I just reverted the old code.
I will let someone with more experience answer this question. :)

I can just guess and say that
maybe the ProcessTransport close method should be modified:
```diff
    def close(self) -> None:
        if not self._closed:
            self._send_queue.put_nowait(None)
            if self._socket:
                self._socket.close()
            self._closed = True
+           self._reader.close()
+           self._writer.close()
+.          self._reader = None
+.          self._writer = None
```

Reply:
Don't know about stdio and TCP, but the multiprocessing pipe is garbage-collectable: https://docs.python.org/3/library/multiprocessing.html#multiprocessing.connection.Connection.close